### ISSUE TITLE
AWS S3ConfigurationProperties are not applied

### ISSUE THREAD

**Issue by Vei0vis** [micronaut-projects/micronaut-aws#1880]
### Expected Behavior

Updated the latest micronaut-parent version 4.1.0 and and found that the Configuration Properties for S3ConfigurationProperties (https://micronaut-projects.github.io/micronaut-aws/latest/guide/#io.micronaut.aws.sdk.v2.service.s3.S3ConfigurationProperties) were not applied correctly/ at all since software.amazon.awssdk s3 2.18.0+.

example:
setting aws.s3.path-style-access-enabled in application.yaml to true, is not working.

Workaround is to downgrade to 2.17.295.

### Version

4.1.0

**Comment by rorueda** [comment]
Same problem here. I noticed the problem after upgrading to micronaut 4 (from 3). It seems a problem with the code generated when using the `@ConfigurationBuilder` as it generates code that calls the builder's getters instead of its setters.

Micronaut 4
```
class $S3ConfigurationProperties$Definition extends AbstractInitializableBeanDefinition<S3ConfigurationProperties> {
  ...
  public Object inject(BeanResolutionContext var1, BeanContext var2, Object var3) {
    ...
      Optional var9 = this.getValueForPath(var1, var2, Argument.of(Boolean.class, "pathStyleAccessEnabled"), "aws.s3.path-style-access-enabled");
      if (var9.isPresent() && (Boolean)var9.get()) {
        try {
          var4.getBuilder().pathStyleAccessEnabled();
        } catch (NoSuchMethodError var22) {
        }
      }
    ...
  }
  ...
}
```

Micronaut 3 
```
class $S3ConfigurationProperties$Definition extends AbstractInitializableBeanDefinition<S3ConfigurationProperties> implements BeanFactory<S3ConfigurationProperties> {
  ...
  protected Object injectBean(BeanResolutionContext var1, BeanContext var2, Object var3) {
    ...
      Optional var6 = this.getValueForPath(var1, var2, Argument.of(Boolean.class, "path-style-access-enabled"), "aws.s3.path-style-access-enabled");
      if (var6.isPresent()) {
        try {
          var4.getBuilder().pathStyleAccessEnabled((Boolean)var6.get());
        } catch (NoSuchMethodError var17) {
        }
      }
    ...
  }
  ...
}
```

**Comment by rorueda** [comment]
I opened an issue on micronaut-core project: https://github.com/micronaut-projects/micronaut-core/issues/9919

**Comment by timyates** [comment]
@sdelamo looks like this _may_ be fixed by https://github.com/micronaut-projects/micronaut-core/pull/9942 which was merged into the 4.2.x branch of Micronaut.

**Comment by sdelamo** [comment]
@timyates can you write a test for this and verify it is fixed and close the issue.

**Comment by sdelamo** [comment]
@Vei0vis it is fixed in Micronaut Framework 4.2.x
